File: application/src/main/java/whenyourcar/application/facade/user/UserCommonFacade.java
package whenyourcar.application.facade.user;

import whenyourcar.domain.entity.User;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import whenyourcar.application.converter.user.UserCommonConverter;
import whenyourcar.application.dto.user.common.UserCommonResponse;
import whenyourcar.application.service.user.UserCommonService;


@Service
@RequiredArgsConstructor
public class UserCommonFacade {
    private final UserCommonService userCommonService;
    private final UserCommonConverter userCommonConverter;
    public UserCommonResponse.UserProfileResponseDto getUserProfile(String email)  {
        User user = userCommonService.getUserByEmail(email);
        return userCommonConverter.toUserProfileResponse(user);
    }
}

File: application/src/main/kotlin/howmuchcar/application/facade/user/UserCommonFacade.kt
package howmuchcar.application.facade.user

import howmuchcar.application.converter.user.UserCommonConverter
import howmuchcar.application.dto.user.UserProfileResponse
import howmuchcar.domain.entity.User
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class UserCommonFacade (
    private val userCommonConverter: UserCommonConverter
){
    fun getUserProfile(user: User) : UserProfileResponse {
        return  userCommonConverter.toUserProfileResponse(user)
    }
}